pennylane/qnodes/base.py

```diff
@@ -314,48 +298,39 @@ def _op_descendants(self, op, only):
             return list(itertools.filterfalse(_is_observable, succ))
         return succ
 
-    def _remove_op(self, op):
-        """Remove a quantum operation from the circuit queue.
-
-        Args:
-            op (:class:`~.operation.Operation`): quantum operation to be removed from the circuit
-        """
-        self.queue.remove(op)
-
-    def _append_op(self, op):
-        """Append a quantum operation into the circuit queue.
-
-        Args:
-            op (:class:`~.operation.Operation`): quantum operation to be added to the circuit
+    def _remove_operator(self, operator):
+        if isinstance(operator, Observable) and operator.return_type is not None:
+            self.obs_queue.remove(operator)
+        else:
+            self.queue.remove(operator)
 
-        Raises:
-            ValueError: if `op` does not act on all wires
-            QuantumFunctionError: if state preparations and gates do not precede measured observables
-        """
-        if op.num_wires == Wires.All:
-            if set(op.wires) != set(range(self.num_wires)):
-                raise QuantumFunctionError("Operator {} must act on all wires".format(op.name))
+    def _append_operator(self, operator):
+        if operator.num_wires == Wires.All:
+            if set(operator.wires) != set(range(self.num_wires)):
+                raise QuantumFunctionError(
+                    "Operator {} must act on all wires".format(operator.name)
+                )
 
         # Make sure only existing wires are used.
-        for w in _flatten(op.wires):
+        for w in _flatten(operator.wires):
             if w < 0 or w >= self.num_wires:
                 raise QuantumFunctionError(
                     "Operation {} applied to invalid wire {} "
-                    "on device with {} wires.".format(op.name, w, self.num_wires)
+                    "on device with {} wires.".format(operator.name, w, self.num_wires)
                 )
 
         # observables go to their own, temporary queue
-        if isinstance(op, Observable):
-            if op.return_type is None:
-                self.queue.append(op)
+        if isinstance(operator, Observable):
+            if operator.return_type is None:
+                self.queue.append(operator)
             else:
-                self.obs_queue.append(op)
+                self.obs_queue.append(operator)
         else:
             if self.obs_queue:
                 raise QuantumFunctionError(
                     "State preparations and gates must precede measured observables."
                 )
-            self.queue.append(op)  # TODO rename self.queue to self.op_queue
```

Comment:
Might as well leave the TODO in?

Reply:
Yes, I think I'll have to do a renaming PR soon!